### DISCUSSION TITLE
How can I embed subtitle For yt-dlp supported links?

### DISCUSSION THREAD

**Discussion by erickyun** [anasty17/mirror-leech-telegram-bot#1300] [Q&A] [answered]
İs it something like? 
`/yl URL opt: EmbedSubtitle:true`
Or
`/yl opt: writesubtitles:true|allsubtitles:true |merge_output_format:"ass"`

How can I do this?

**Comment by anasty17** [comment, marked as answer]
Check the yt-dlp option it's in /y. U will see hyperlink there. U can check all options and there is an examples. Yes the format is right of ur example but idk if those settings are correct or not

**Reply by erickyun** [reply]
Thank you for your response. I did read from there but I couldn't find the right cmd? İt would be great if someone could give me example? :)

**Reply by anasty17** [reply]
Hmmm I can do it for u but i don't have enough time. Why u don't join yt-dlp support chat on discord.

**Reply by erickyun** [reply]
Thanks I will try that but could you still write it here if found the cmd of how to embed subtitles? 
So it will also help others. :)

**Reply by erickyun** [reply]
> Hmmm I can do it for u but i don't have enough time. Why u don't join yt-dlp support chat on discord.

I got such answer if it would be useful to you?
https://github.com/yt-dlp/yt-dlp/issues/5859#issuecomment-1363938900

> Here's some useful code that translates b/w CLI and API options
> 
> ```python
> import yt_dlp
> 
> def cli_to_api(*opts):
>     default = yt_dlp.parse_options([]).ydl_opts
>     diff = {k: v for k, v in yt_dlp.parse_options(opts).ydl_opts.items() if default[k] != v}
>     if 'postprocessors' in diff:
>         diff['postprocessors'] = [pp for pp in diff['postprocessors'] if pp not in default['postprocessors']]
>     return diff
>     
> from pprint import pprint
> 
> pprint(cli_to_api('-4', '--downloader', 'http,m3u8:aria2c', '--embed-thumb'))  # Change according to your need
> ```
> 
> So, to answer your question,
> 
> ```python
> >>> cli_to_api('--embed-sub')
> {'writesubtitles': True, 'postprocessors': [{'key': 'FFmpegEmbedSubtitle', 'already_have_subtitle': False}]}
> ```

**Reply by anasty17** [reply]
Update ur code and add this opt:

opt: writesubtitles:True|postprocessors: [{'key': 'FFmpegEmbedSubtitle', 'already_have_subtitle': False}]

and pleased don't post teo same comments in different issues.

**Reply by erickyun** [reply]
> Update ur code and add this opt:
> 
> opt: writesubtitles:True|postprocessors: [{'key': 'FFmpegEmbedSubtitle', 'already_have_subtitle': False}]

İt is not working? :(
I get this error? 
```dictionary update sequence element #0 has length 1; 2 is required```

**Reply by anasty17** [reply]
Update ur code and try

**Reply by erickyun** [reply]
Yes it working fine now.👍👍👍☺️
Thanks for helping me everyone.

```
/yzl s opt: final_ext:mkv|writesubtitles:True|allsubtitles:true|postprocessors: [{'key': 'FFmpegVideoRemuxer', 'preferedformat': 'mkv'}, {'key': 'FFmpegEmbedSubtitle', 'already_have_subtitle': False}]
```
